File: ItCameDroid-AS/app/src/main/java/br/odb/giovanni/game/Vec2.java
package br.odb.giovanni.game;

public class Vec2 {
	public float x;
	public float y;

	public Vec2(float aX, float aY) {
		x = aX;
		y = aY;
	}

	public Vec2() {
		x = 0;
		y = 0;
	}

	@Override
	public boolean equals(Object o) {
		if (o instanceof Vec2) {
			Vec2 v = (Vec2) o;
			return v.x == x && v.y == y;
		} else {
			return false;
		}
	}

	public boolean isCloseEnoughToConsiderEqualTo(Vec2 position) {
		return (Math.abs(x - position.x) < 1.0f) && (Math.abs(y - position.y) < 1.0f);
	}
}

File: ItCameDroid-AS/app/src/main/java/br/odb/giovanni/game/Vec2.kt
package br.odb.giovanni.game

import kotlin.math.abs

class Vec2 {
    @JvmField
	var x: Float
    @JvmField
	var y: Float

    constructor(aX: Float, aY: Float) {
        x = aX
        y = aY
    }

    constructor() {
        x = 0f
        y = 0f
    }

    override fun equals(o: Any?): Boolean {
        return if (o is Vec2) {
            o.x == x && o.y == y
        } else {
            false
        }
    }

    fun isCloseEnoughToConsiderEqualTo(position: Vec2): Boolean {
        return abs(x - position.x) < 1.0f && abs(y - position.y) < 1.0f
    }
}